src/scanpy/tools/_score_genes.py

```diff
@@ -161,14 +165,29 @@ def get_subset(genes: pd.Index[str]):
 
     n_items = int(np.round(len(obs_avg) / (n_bins - 1)))
     obs_cut = obs_avg.rank(method="min") // n_items
-    control_genes = pd.Index([], dtype="string")
+    obs_cut_is_ctrl = False if ctrl_as_ref else obs_cut.index.isin(gene_list)
```

Comment:
Not sure if I fully understand the naming here. I might go with something like `keep_ctrl_in_obs_cut`

Reply:
great name, much better!